src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,141 @@
+package duke;
+
+import duke.task.Task;
+import duke.task.TaskList;
+import java.util.Scanner;
+
+/**
+ * Represents the user interface for Duke, handling input and output.
+ */
+public class Ui {
+    private Scanner scanner;
+
+    /**
+     * Constructs a Ui object and initializes the scanner for user input.
+     */
+    public Ui() {
+        scanner = new Scanner(System.in);
+    }
+
+    /**
+     * Displays a welcome message when Duke starts and returns it as a String.
+     */
+    public String showWelcome() {
+        String logo = " ____        _        \n"
+                + "|  _ \\ _   _| | _____ \n"
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        String welcomeMessage = "Hello from\n" + logo + "\n"
+                + "\t____________________________________________________________\n"
+                + "\t Hello! I'm Duke.\n"
+                + "\t What can I do for you?\n"
+                + "\t____________________________________________________________";
+        return welcomeMessage;
+    }
+
+    /**
+     * Displays an exit message when Duke is exited and returns it as a String.
+     */
+    public String showExit() {
+        return "\t Bye. Hope to see you again soon!";
+    }
+
+    /**
+     * Displays an error message when loading tasks from storage fails and returns it as a String.
+     */
+    public String showLoadingError() {
+        return "No saved tasks found.";
+    }
+
+    /**
+     * Displays an error message with the provided error message and returns it as a String.
+     *
+     * @param errorMessage The error message to display.
+     */
+    public String showError(String errorMessage) {
+        return errorMessage;
+    }
+
+    /**
+     * Returns a horizontal line separator as a String.
+     */
+    public String showLine() {
```

Comment:
Perhaps store the line as a class attribute?

```suggestion
    private final String LINE = "-".repeat(60);
```